app/models/ticket_request.rb

```diff
@@ -270,23 +275,20 @@ def refund
     if refunded?
       errors.add(:base, 'Cannot refund a ticket that has already been refunded')
       return false
+    elsif !completed? || !payment&.refundable?
+      errors.add(:base, 'Cannot refund a ticket that has not been purchased')
+      return false
     end
 
-    return if payment&.received?
-
-    errors.add(:base, 'Cannot refund a ticket that has not been purchased')
-    false
-
-    # XXX need to build refund. Put into Payment model
-    # begin
-    #   TicketRequest.transaction do
-    #     Stripe::Charge.retrieve(payment.stripe_charge_id).refund
-    #     return update(status: STATUS_REFUNDED)
-    #   end
-    # rescue Stripe::StripeError => e
-    #   errors.add(:base, "Cannot refund ticket: #{e.message}")
-    #   false
-    # end
+    # issue refund for payment
+    Rails.logger.info { "ticket_request [#{id}] payment [#{payment.id}] refunding [#{payment.stripe_payment_id}]" }
+    if payment.refund_payment
+      mark_refunded
+      true
+    else
+      Rails.logger.error { "ticket_request failed to refund [#{payment.stripe_payment_id}]" }
```

Comment:
Please add the cause to the logger, by adding `errors&.error_messages&.join('; ')`.